ts-compilation-test/tsconfig.json

```diff
@@ -6,7 +6,7 @@
     "outDir": "lib",
     "sourceMap": true,
     "strict": true,
-    "target": "es2017"
+    "target": "es2018"
```

Comment:
I had to bump this to get the sample app to compile with TS properly, so this may be an issue if we want to include the `AsyncGenerator` stuff.

This also makes me question the `tsconfig` for the TS app itself, I may need to dig into this a little if we want to keep the generator included. 🤔

Reply:
Discussed offline: Given this may disrupt TypeScript consumers of the library even with an easy fix on their end, it seems prudent to pull this functionality out for now and wait until a major version bump.